app/services/themes/theme_site_setting_manager.rb

```diff
@@ -0,0 +1,140 @@
+# frozen_string_literal: true
+
+# Responsible for creating or updating theme site settings, and in the case
+# where the value is set to nil or is the same as the site setting default,
+# deleting the theme site setting override.
+#
+# Theme site settings are used to override specific site settings that are
+# marked as themeable in site_settings.yml. This allows themes to have a greater
+# control over the full site experience.
+#
+# Theme site settings have an identical schema to SiteSetting.
+class Themes::ThemeSiteSettingManager
+  include Service::Base
+
+  params do
+    attribute :theme_id, :integer
+    attribute :name
+    attribute :value, :string
+
+    validates :theme_id, presence: true
+    validates :name, presence: true
+
+    after_validation do
+      self.name = self.name.to_sym if self.name.present?
+      self.value = self.value || nil
```

Comment:
Is it correct? If the value is `false`, it would be converted to `nil`. Maybe we should keep `false`, so `false` is overridden by user and `nil` means - delete this override.